plugins/modules/ec2_vpc_nat_gateway.py

```diff
@@ -633,10 +639,8 @@ def create(client, module, subnet_id, allocation_id, tags, purge_tags, client_to
                 _describe_nat_gateways(client, NatGatewayIds=[result['nat_gateway_id']])[0]
             )
 
-        result['tags'], _tags_update_exists = ensure_tags(
-            client, module, nat_gw_id=result['nat_gateway_id'], tags=tags,
-            purge_tags=purge_tags
-        )
+        changed |= add_ec2_tags(client, module, result['nat_gateway_id'], tags)
```

Comment:
Since botocore 1.15.17 we can pass TagSpecifications to create_nat_gateway and do the tagging as part of the creation.

See for example https://github.com/ansible-collections/amazon.aws/pull/522/files#diff-600cb6bef37f4a5322ddd42545c971d923dd24e02604ee147d15e814995124d3R407-R411